mooncake-store/src/master_service.cpp

```diff
@@ -460,6 +464,18 @@ auto MasterService::PutRevoke(const std::string& key, ReplicaType replica_type)
                    << ", error=invalid_replica_status";
         return tl::make_unexpected(ErrorCode::INVALID_WRITE);
     }
+    // When disk replica is enabled, update allocated_file_size
+    if (use_disk_replica_ && replica_type == ReplicaType::DISK) {
```

Comment:
How about we do it in ObjectMetadata's constructor and destructor? That's more RAII and the code looks cleaner.